app/controllers/catalog_controller.rb

```diff
@@ -613,27 +617,18 @@ def ot_copy
 
   def ot_edit_submit
```

Comment:
`ot_add_submit`, `ot_edit_submit` and `ot_copy_submit` should all be dead now, right? (And the routes.)

Those were called by 

```
      elsif %w[ot_edit ot_copy ot_add service_dialog_from_ot].include?(action)
...
                  :action_url => "#{action}_submit",
```

in `CatalogController#replace_right_cell`. Which probably means that branch should no longer apply for the 3?

Reply:
So there is something strange going on. If i delete the routes the haml template wont change. I also want to remove the create service dialog from OT and has the same problem(its waiting for API changes). 

So if i delete the line i should be able to delete the routes?

Reply:
We are talking about the ot_add/edit/copy form templates. It throws an error if i delete the routes. I will send you more details about it once i have a bit more time (Thursday/Friday)

Reply:
Ah, yes, calling url_for with that action will fail once you remove that action.

But why do you want to call a haml that computes an url you won't use? That's the bug.

Reply:
nvm i know whats the issue. Just needed to delete one of many else if statements in replace right cell. It was trying to set the action for the buttons, but since they are not used anymore i we can just delete it.


```diff
diff --git a/app/controllers/catalog_controller.rb b/app/controllers/catalog_controller.rb
index 5640fba92..dd3459e91 100644
--- a/app/controllers/catalog_controller.rb
+++ b/app/controllers/catalog_controller.rb
@@ -615,18 +615,6 @@ class CatalogController < ApplicationController
     replace_right_cell(:action => "ot_copy")
   end
 
-  def ot_edit_submit
-    self.x_active_tree = 'ot_tree'
-  end
-
-  def ot_add_submit
-    self.x_active_tree = 'ot_tree'
-  end
-
-  def ot_copy_submit
-    self.x_active_tree = 'ot_tree'
-  end
-
   def ot_form_field_changed
     dialog_creation_form_field_changed("ot_edit__#{params[:id]}")
   end
@@ -1876,17 +1864,6 @@ class CatalogController < ApplicationController
             ]
           )
         end
-      elsif %w[ot_edit ot_copy ot_add service_dialog_from_ot].include?(action)
-        presenter.hide(:toolbar).show(:paging_div, :form_buttons_div).remove_paging
-        locals = {:record_id  => @edit[:rec_id],
-                  :action_url => "#{action}_submit",
-                  :serialize  => true}
-        if action == "ot_copy"
-          presenter.show(:buttons_on).hide(:buttons_off)
-          locals[:record_id] = nil
-        end
-        locals[:no_reset] = true if %w[ot_copy service_dialog_from_ot].include?(action)
-        presenter.update(:form_buttons_div, r[:partial => "layouts/x_edit_buttons", :locals => locals])
       else
         # Added so buttons can be turned off even tho div is not being displayed it still pops up Abandon changes box when trying to change a node on tree after saving a record
         presenter.hide(:buttons_on, :form_buttons_div).show(:toolbar).hide(:paging_div)

```

Reply:
what about that `service_dialog_from_ot`?

Reply:
Oh right that should stay. (I am working in that in different PR though. Need some API changes unfortunately)

Reply:
Ok should be fine now.  I've added `locals[:no_reset]`  to the initial locals hash and is always true because now it works only for the service_dialog_from_ot action. I think that there is nothing more i can delete without breaking anything.